### ISSUE TITLE
typedefs for functions with varargs do not work

### ISSUE THREAD

**Issue by gmp216** [elliotchance/c2go#667]
The following example program causes an error during transpilation:
```
#include <stdio.h>

typedef void function_t (va_list args);

int main() {
}
```
This results in the following error:

> panic: unknown node type: 'DecayedType 0x55646dc19b80 'struct __va_list_tag *' sugar'

Tested and failed on both MacOS and Linux.

**Comment by elliotchance** [comment]
Thank you for raising this bug @gmp216. This error means that c2go can not understand one of lines from the clang output.

If you would like to have a go (no pun intended) at fixing this, you can duplicate the `ast/aligned_attr.go` and `ast/aligned_attr_test.go` files and call them `ast/decayed_type.go` and `ast/decayed_type_test.go` respectively. You should end up with one test case (using the panic message above) and modify `ast/decayed_type.go` to work for that case.

If you have any questions or run into any issues, just jump on our chat: https://gitter.im/c2goproject/Lobby

**Comment by gmp216** [comment]
Cool, thanks for the help, should be fixed now (see #668).
